File: javalin/src/main/java/io/javalin/core/compression/Gzip.java
package io.javalin.core.compression;

public class Gzip {

    private final int level;

    /**
     * Create with default compression level of 6.
     */
    public Gzip() {
        this.level = 6;
    }

    /**
     * @param level Compression level. Higher yields better (but slower) compression. Range 0..9, default = 6
     */
    public Gzip(int level) {
        if (level < 0 || level > 9) {
            throw new IllegalArgumentException("Valid range for parameter level is 0 to 9");
        }
        this.level = level;
    }

    public int getLevel() {
        return level;
    }
}

File: src/main/java/io/javalin/core/compression/Gzip.kt
package io.javalin.core.compression

/**
 * @param level Compression level. Higher yields better (but slower) compression. Range 0..9, default = 6
 */
class Gzip(val level: Int = 6) {
    init {
        require(level in 0..9) { "Valid range for parameter level is 0 to 9" }
    }
}
